src/tests/serde.rs

```diff
@@ -594,3 +600,188 @@ fn test_serialize_deserialize_unsigned_numbers() {
     let doc_result: Result<Document, serde_json::Error> = serde_json::from_str(&json);
     assert!(doc_result.is_err());
 }
+
+#[test]
+fn test_unsigned_helpers() {
+    let _guard = LOCK.run_concurrently();
+
+    #[derive(Serialize)]
+    struct A {
+        #[serde(serialize_with = "serde_helpers::serialize_u32_as_i32")]
+        num_1: u32,
+        #[serde(serialize_with = "serde_helpers::serialize_u64_as_i32")]
+        num_2: u64,
+    }
+
+    let a = A { num_1: 1, num_2: 2 };
+    let doc = to_document(&a).unwrap();
+    assert!(doc.get_i32("num_1").unwrap() == 1);
+    assert!(doc.get_i32("num_2").unwrap() == 2);
+
+    let a = A {
```

Comment:
I think this should be split into two separate failure cases, one where only `num_1` is too high and the other where only `num_2` is. As it's written right now, assuming `serialize_u2_as_i32` works properly, I don't think `serialize_u64_as_i32` is being tested.